analyses/oncoprint-landscape/util/oncoplot-functions.R

```diff
@@ -0,0 +1,103 @@
+# This script defines custom functions to be sourced in the
+# `01-plot-oncoprint.R` script of this module.
+#
+# Chante Bethell for CCDL 2020
+#
+# # #### USAGE
+# This script is intended to be sourced in the script as follows:
+#
+# source(file.path("util", "oncoplot-functions.R"))
+
+prepare_and_plot_oncoprint <- function(maf_df,
+                                       cnv_df,
+                                       metadata,
+                                       fusion_df = NULL,
+                                       goi_list = NULL,
+                                       color_palette,
+                                       running_in_figures_script = FALSE) {
+  # Given maf, cnv, and fusion data.frames prepared in `00-map-to-sample_id.R`,
+  # along with a list of genes prepared in the `interaction-plots` directory,
+  # plot an oncoprint landscape figure.
+  #
+  # Args:
+  #   maf_df: data.frame with data from a MAF file
+  #   cnv_df: data.frame with copy number variant data
+  #   metadata: data.frame with the relavant metadata
+  #   fusion_df: data.frame with fusion data. This is NULL by default.
+  #   goi_list: a list or vector of genes of interest that should be
+  #             represented on the oncoprint. This is NULL by default.
+  #   color_palette: vector of colors (hex codes) corresponding to the data
+  #                  that will be represented on the oncoprint
+  #   running_in_figures_script: TRUE or FALSE statement indicating whether
+  #                              or not this function is being run in the
+  #                              figures generation script
+  
+  if (!is.null(fusion_df)) {
+    # Bind rows of maf and fusion data frames
+    maf_df <- dplyr::bind_rows(maf_df, fusion_df)
```

Comment:
Yes it does create NAs in the data frame. `fusion_df` has 5 columns, 4 of which overlap with `maf_df` which has 135 columns.

I'll also note that `fusion_df` is significantly smaller than `maf_df` so I am not sure what you initially meant here by _a lot_ of NAs but when running this using the subset file, the `fusion_df` data accounts for 32 of the 12147 observations in the data frame created using `bind_rows`. In other words, there are 32 rows that contain NAs.

Reply:
It might be best to be a bit more intentional about the columns being bound so that way if something fails, we get a warning kicked back. In other words, what columns do you need downstream? Can we do a `select` of those columns and then bind the `fusion_df` and `maf_df` of just those columns? This way we can also double check what information is being passed so we can construct this in a way to know if we are missing information at some point.